[junlin] iP

chanjunlin opened the pull request:
ChinChin

> “Your mind is for having ideas, not holding them.” – David Allen ([source](https://dansilvestre.com/productivity-quotes))

ChinChin frees your mind of having to remember things you need to do! 😄 
It is:
- Text-based
- Easy to learn
- ~~FAST~~ _SUPER_ FAST to use

Features included are:
- Adding / deleting tasks
- Marking / Un-marking tasks
- Local memory update -> Everything you key in would be stored!

All you need to do is,
1. download it from [here](https://github.com/nus-cs2103-AY2425S2/ip/pull/361).
2. double-click it.
3. add your tasks
4. let it manage your tasks for you 😉 

Upcoming features:
- [ ] Managing deadlines 
- [ ] Reminders

Commit:
Add Gradle support

Eclipse-Dominator committed:
Bump gradle and lib version

aureliony committed:
build.gradle: Prevent generating a second JAR file

In build.gradle, the dependencies on distZip and/or distTar causes
the shadowJar task to generate a second JAR file for which the
mainClass.set("seedu.duke.Duke") does not take effect.
Hence, this additional JAR file cannot be run.
For this product, there is no need to generate a second JAR file
to begin with.

Let's remove this dependency from the build.gradle to prevent the
shadowJar task from generating the extra JAR file.

chanjunlin committed:
Changed bot name

chanjunlin committed:
Echo

chanjunlin committed:
Add, list

chanjunlin committed:
Add task management feature and changed user input handling

- Changed ChinChin.java to simplify input handling to use switch-case
- Changed CustomList.java to manage task states and display formatted output
- Changed CUstomList.java from ArrayList<String> to ArrayList<Task>
- Added feature for user to "mark <number>" and "unmark <number>"
- Added Task.Java for a task class

chanjunlin committed:
Added support for tracking Todo, Deadline and Event

- Renamed Task.java to Todo.java
- Modified Todo Class to display the Task's tag at the front
	- [T] for Todo
	- [D] for Deadline
	- [E] for Event
- Created Deadline class which inherits from Todo Class for tasks with deadlines
- Created Event class which inherits from Todo Class for Events

chanjunlin committed:
Automating the testing of text UIs

- Updated input.txt to contain sample user commands
- Updated EXPECTED.txt to capture expected outputs based on corresponding inputs
- Updated runtest.sh to test the inputs properly

chanjunlin committed:
Handling errors

- Created ChinChinException.java for custom exceptions
- Modified ChinChin.java to capture incorrect inputs by user, i.e:
	- No task's description
	- Deadline task with no deadline
	- Event task with no "from" and "to"
	- "mark" and "unmark" followed by a non-integer input

chanjunlin committed:
Add support for deleting task from the list

- Modified ChinChin.java to allow user to "delete" task
- Modified CustomList.java to support the deletion of task
- Modified Event.java to improve the UI

chanjunlin committed:
Refactor task tag implementation to use TaskType enum

Task tags are currently implemented as hardcoded strings in the code ([T], [D], [E]).

Hardcoded tags makes the code harder to maintain or extend in future.

Switch tagging mechanisms across all tasks ('Task', 'Deadline', 'Event') to use enumeration called TaskType.

This improves code clarity, reduces duplication and makes future extensions easier.

Replace hardcoded task tags with an enumeration ('TaskType')

chanjunlin committed:
Add Storage class and refactor existing Java files for improvements

- New class
	- Added a new 'Storage.class' for managing task data storage and retriefval in the application.

- Refactoring
	- Updated 'ChinChin.java' to utilise the new Storage class for improved modularity.
	- Enhanced 'CustomList.java' to ensure that all task modifications are reflected.

These changes improve code organisation, readability and adheres to the Java coding standards.

chanjunlin committed:
Merge branch 'branch-Level-7' into master to intergrate saving functionality.

This merge integrates the newly developed 'Storage' class into the master branch.

chanjunlin committed:
Add DateFormatter class and improve task, deadline and event parsing

- New Class:
	- Create 'DateFormatter' that holds all possible accepted formats for user input.
	- Can be edited to add more formats

- Improve parsing:
	- Update 'Deadline' and 'Event' class to utilise the new 'DateFormatter' for parsing dates.
	- Convert the input strings into actual 'LocalDateTime' objects instead of treating them as plaintext.

chanjunlin committed:
Merge branch 'branch-Level-8'

chanjunlin committed:
Refactor codebase to improve OOP design and extract classes

- Extract 'ChinChinUI' class for managing user interactions.
- Extract 'Storage' class for handling file I/O operations,
- Extract 'ChinChinParser' class for parsing user commands into executable actions.
- Create abstract 'ChinChinCommand' base class with child classes:
	- "AddCommand", "DeleteCommand", "ExitCommand", "ListCommand", etc.
	- Simplified command execution logic through polymorphism.
- Refactor 'ChinChin' class to align with the new structure.

chanjunlin committed:
Merge branch 'A-MoreOOP'

chanjunlin committed:
Reorganize project structure into packages

- Group related classes into logical Java packages:
  - Move main application-related files (e.g., ChinChin, ChinChinException) into `chin.main`.
  - Move user interface-related files (e.g., ChinChinUI) into `chin.ui`.
  - Move task-related files (e.g., Task, Deadline, Event, TaskType) into `chin.task`.
  - Move command classes (e.g., AddCommand, DeleteCommand, ExitCommand) into `chin.command`.
  - Move files handling logic (Storage.java) into `chin.storage`.
  - Move util files (ChinChinParser.java) into `chin.util`.

chanjunlin committed:
Merge branch 'A-Packages'

RickyHFR commented on src/main/java/ChinChin/command/AddCommand.java:
```diff
@@ -0,0 +1,53 @@
+        } else {
+            throw new ChinChinException("Huh..");
+        }
+    }
```
String taskInfo = taskList.todoTask(userInput); is repeated for each type of task. You could create a helper function to avoid repeating this in each if block. taskListSize is also calculated multiple times. It could be moved outside of the conditional blocks to reduce redundancy.

RickyHFR commented on src/main/java/ChinChin/command/AddCommand.java:
```diff
@@ -0,0 +1,53 @@
+     */
+    @Override
+    public void execute(CustomList taskList, ChinChinUI chinChinUI, Storage storage) throws ChinChinException {
+        if (userInput.startsWith("todo") || userInput.startsWith("Todo")) {
```
You are checking the case with userInput.startsWith("todo") || userInput.startsWith("Todo"). It might be clearer and more concise to convert the user input to lowercase (e.g., userInput.toLowerCase().startsWith("todo")). This way, you only need to check for one case (e.g., todo) and improve maintainability.

RickyHFR commented on src/main/java/ChinChin/main/ChinChin.java:
```diff
@@ -0,0 +1,58 @@
+     * @param args Command line arguments passed to the program
+     */
+    public static void main(String[] args) {
+        new ChinChin("data/ChinChinTaskList.txt").run();
```
You may consider [specify file paths in an OS-independent way](https://www.sghill.net/how-do-i-make-cross-platform-file-paths-in-java.html) to make your app work when used on a different OS.

RickyHFR commented on src/main/java/ChinChin/task/Deadline.java:
```diff
@@ -0,0 +1,53 @@
+ * Represent a DEADLINE task with a due date
+ */
+public class Deadline extends Task {
+    private LocalDateTime dueDate;
```
The name of variable is very clear:)

RickyHFR commented on src/main/java/ChinChin/task/Event.java:
```diff
@@ -0,0 +1,59 @@
+ */
+public class Event extends Task {
+    private LocalDateTime starting;
+    private LocalDateTime ending;
```
```suggestion
    private LocalDateTime startingDate;
    private LocalDateTime endingDate;
```

You may consider adding a Date to make the variable name easier to understand:)

RickyHFR reviewed (commented):
LGTM. Overall your code quality is very solid. Error handling is super comprehensive as well!

chanjunlin committed:
Merge branch 'A-Gradle'

chanjunlin committed:
Change build.gradle to run ChinChin.java

chanjunlin committed:
Add JUnit Tests

chanjunlin committed:
Add functionality to search task by keywords

Application currently does not support users to locate tasks based on specific keywords.

To improve usability:
- Implement 'FindCommand' that allow user to search through their tasks list with specific keywords.
- Display all tasks with their corresponding index number if found.
- Inform the user if no matching tasks exist in the task list.

chanjunlin committed:
Modify the index to show the correct index

chanjunlin committed:
Merge tag 'Level-9'

chanjunlin committed:
Add comment to merge Branch-Level-9

chanjunlin committed:
Merge branch 'Branch-Level-9'

chanjunlin committed:
Added comment to merge with Branch-A-CodingStandard

chanjunlin committed:
Merge branch 'Branch-A-CodingStandard'

chanjunlin committed:
Add comment to merge with Branch-A-JavaDoc

chanjunlin committed:
Merge branch 'Branch-A-JavaDoc'

Reshiro commented on src/main/java/ChinChin/main/ChinChin.java:
```diff
@@ -0,0 +1,72 @@
+    }
+
+    /**
+     * Reinitialises the lists of tasks from storage
```
Be sure to use American spelling (initialize) for comments.

Reshiro commented on src/main/java/ChinChin/command/ChinChinCommand.java:
```diff
@@ -0,0 +1,31 @@
+package ChinChin.command;
+
+import ChinChin.command.*;
```
IntelliJ might have automatically added wildcard imports here, I believe there's a setting to turn that off.

Reshiro commented on src/main/java/ChinChin/command/ChinChinCommand.java:
```diff
@@ -0,0 +1,31 @@
+    /**
+     * Executes the command
+     *
+     * @param taskList The customList holding all the tasks
```
Punctuation should be used behind each parameter description.

Reshiro commented on src/main/java/ChinChin/storage/Storage.java:
```diff
@@ -0,0 +1,154 @@
+
+        try {
+            switch (command) {
+            case ("todo"):
```
Good job on the proper indents for switch statements.

Reshiro commented on src/main/java/ChinChin/task/Task.java:
```diff
@@ -0,0 +1,79 @@
+     *
+     * @return A boolean value depending if the task is done or not
+     */
+    public boolean isDone() {
```
Method name should be a verb, it doesn't need to have boolean naming even though it returns a boolean.

Reshiro commented on src/main/java/ChinChin/util/DateFormatter.java:
```diff
@@ -0,0 +1,40 @@
+            DateTimeFormatter.ofPattern("M/d/yyyy HHmm"),
+            DateTimeFormatter.ofPattern("MM/dd/yyyy HHmm"),
+            DateTimeFormatter.ofPattern("yyyy-MM-dd HHmm"),
+//            DateTimeFormatter.ofPattern("yyyy-MM-dd HHmm"),
```
I think you can remove these comments. I assume they're supposed to be temporary?

Reshiro reviewed (commented):
The code adheres to the coding standard pretty well, there's just a few minor violations with comments and method naming.

chanjunlin committed:
Refactor Java files for Checkstyle compliance

- Rename packages from 'ChinChin' to 'chin' to comply with Java naming conventions.
- Update import statements to avoid wildcard imports and ensure correct order.
- Address various style violations detected by Checkstyle.
- Modify build.gradle as needed to reflect new package paths.

chanjunlin committed:
Merge tag 'A-CheckStyle'

chanjunlin committed:
Make the branch-A-JavaDoc green in the iP dashboard

chanjunlin committed:
Merge branch 'branch-A-JavaDoc'

chanjunlin committed:
Make 'branch-A-CodingStandard' green in iP dashboard

chanjunlin committed:
Make 'branch-Level-9' green in iP dashboard

chanjunlin committed:
Merge branch 'branch-Level-9'

chanjunlin committed:
Add JavaFX GUI for ChinChin

- Implement basic user interface for JavaFX.

chanjunlin committed:
Rectify merging conflicts

chanjunlin committed:
Merge branch 'branch-A-CodingStandard'

chanjunlin committed:
Improve code quality

- Refactor large methods into smaller methods to follow SLAP
- Remove redundant comments and code

chanjunlin committed:
Merge branch 'branch-A-CodeQuality'

chanjunlin committed:
Merge branch 'master' into branch-A-Assertions

chanjunlin committed:
Add assertations for debugging input validation

chanjunlin committed:
Merge branch 'branch-A-Assertions'

chanjunlin committed:
Improve GUI for ChinChin

- Align the words when displaying the list

chanjunlin committed:
Merge branch 'branch-A-BetterGui'

chanjunlin committed:
Refactor the code to improve the GUI

- Modified Java code across multiple files to improve overall functionality and enhance user interface
-

chanjunlin committed:
Merge branch 'branch-A-BetterGui'

chanjunlin committed:
Merge branch 'master' into branch-B-ViewSchedule

chanjunlin committed:
Refactor the code to add View Schedule feature

- Implement a new ViewCommand to filter and display tasks scheduled on a specific date.
- Update the parser to recognise the "view" command and parse user input for date.
- Enhanced UI output to format and dispaly filtered tasks in an organised manner.

chanjunlin committed:
Merge pull request #2 from chanjunlin/branch-B-ViewSchedule

Branch b view schedule

chanjunlin committed:
Refactor the code to improve GUI

- Change font size and color to  make text easier to read.
- Change font to make the text ordered in spacing.

chanjunlin committed:
Merge branch 'branch-B-ViewSchedule'

chanjunlin committed:
Merge pull request #3 from chanjunlin/branch-B-ViewSchedule

Refactor the code to improve GUI

chanjunlin committed:
Update User Guide

- Updated README.md to give information on ChinChin
- Include Ui.png

chanjunlin committed:
Merge branch 'branch-A-UserGuide'

chanjunlin committed:
Refactor ChinChinUI

- Remove all old and redundant code from ChinChinUI.java

chanjunlin committed:
Merge branch 'master' of https://github.com/chanjunlin/ip

chanjunlin committed:
Update README.md to add the description of features

chanjunlin committed:
Merge branch 'branch-A-UserGuide'

chanjunlin committed:
Refactor code to improve README.md

- Add the expected output for each command
- Improve on some bugs

chanjunlin committed:
Merge branch 'branch-A-UserGuide'

chanjunlin committed:
Update GUI

chanjunlin committed:
Merge master into branch-A-BetterGui

chanjunlin committed:
Merge branch 'branch-A-BetterGui'

chanjunlin committed:
Refactor code to improve on the JUnit testing and syntax

- Add more testing features.
- Improve on the syntax of the java code.

chanjunlin committed:
Merge pull request #4 from chanjunlin/branch-A-JUnit

Refactor code to improve on the JUnit testing and syntax

chanjunlin committed:
Merge branch 'master' into branch-B-ViewSchedule

chanjunlin committed:
Refactor the code to improve Gui

- Improve the GUI
- Add "Date" feature explanation in README.md

chanjunlin committed:
Merge branch 'branch-B-ViewSchedule'

chanjunlin committed:
Update README.md to improve readability

chanjunlin committed:
Add JavaDocs comments

chanjunlin committed:
Merge branch 'master' into branch-A-UserGuide

chanjunlin committed:
Refactor README.md

- Update the README.md to improve user readability
- Improve the syntaxes of some description

chanjunlin committed:
Merge branch 'branch-A-UserGuide'

chanjunlin committed:
Improve syntax of README.md

chanjunlin committed:
Merge branch 'branch-A-UserGuide'

chanjunlin committed:
Refactor README.md

- Add 'help' command
- Improve syntax of README.md to improve readability

chanjunlin committed:
Merge branch 'branch-A-UserGuide'

chanjunlin committed:
Update Ui.png

chanjunlin committed:
Refactor README.md to improve readability

chanjunlin committed:
Refactor README.md to improve readability

chanjunlin committed:
Refactor README.md to improve readability

chanjunlin committed:
Refactor README.md to improve readability

chanjunlin committed:
Refactor README.md to improve readability

chanjunlin committed:
Refactor README.md to improve readability
